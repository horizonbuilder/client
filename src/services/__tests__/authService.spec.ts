import AuthService from '../auth';

describe('AuthService', () => {
  describe('#login', () => {
    test('returns true if login was successful', async () => {
      global.fetch = jest.fn(async () => ({
        json: async () => ({ status: 'success' })
      }));

      const loginAttempt = await AuthService.login('test', 'test');

      expect(loginAttempt).toBeTruthy();
    });

    test('returns false if login was not successful', async () => {
      global.fetch = jest.fn(async () => ({
        json: async () => ({ status: 'nope!', error: 'test' })
      }));

      const loginAttempt = await AuthService.login('test', 'test');

      expect(loginAttempt).toMatchSnapshot();
    });
  });

  describe('#signup', () => {
    test('returns true if signup was successful', async () => {
      global.fetch = jest.fn(async () => ({
        json: async () => ({ status: 'success' })
      }));

      const loginAttempt = await AuthService.signup('test', 'test');

      expect(loginAttempt).toMatchSnapshot();
    });

    test('returns false if signup was not successful', async () => {
      global.fetch = jest.fn(async () => ({
        json: async () => ({ status: 'nope!', error: 'test error' })
      }));

      const loginAttempt = await AuthService.signup('test', 'test');

      expect(loginAttempt).toMatchSnapshot();
    });
  });
});
