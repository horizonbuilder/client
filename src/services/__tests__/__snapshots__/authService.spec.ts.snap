// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthService #login returns false if login was not successful 1`] = `
Object {
  "error": "test",
  "status": "nope!",
}
`;

exports[`AuthService #signup returns false if signup was not successful 1`] = `
Object {
  "error": "test error",
  "status": "nope!",
}
`;

exports[`AuthService #signup returns true if signup was successful 1`] = `
Object {
  "status": "success",
}
`;
